src/rnp/rnpcfg.c

```diff
@@ -497,6 +576,105 @@ rnp_cfg_get_defkey(rnp_cfg_t *cfg, rnp_params_t *params)
     } else {
         params->defkey = strdup(userid);
     }
+}
 
-    return RNP_OK;
+/**
+ * @brief Grabs date from the string in %Y-%m-%d format
+ *
+ * @param s [in] NULL-terminated string with the date
+ * @param t [out] On successfull return result will be placed here
+ * @return true on success or false otherwise
+ */
+
+static bool
+grabdate(const char *s, int64_t *t)
+{
+    static regex_t r;
+    static int     compiled;
+    regmatch_t     matches[10];
+    struct tm      tm;
+
+    if (!compiled) {
+        compiled = 1;
+        (void) regcomp(
```

Comment:
We should really be checking these return values, regcomp makes allocations and can fail. I think this was just a function move though, so feel free to dismiss.